File: advent_of_code/2024/01/second.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        string filePath = "input.txt";
        int ans = 0;
        List<int> l1 = new List<int>();
        List<int> l2 = new List<int>();

        try
        {
            // Read all lines from the file into a string array
            string[] lines = File.ReadAllLines(filePath);

            // Print each line
            foreach (string line in lines)
            {
                int a, b;
                string[] parts = line.Split("   ");
                a = int.Parse(parts[0]);
                l1.Add(a);
                b = int.Parse(parts[1]);
                l2.Add(b);
            }
        }
        catch (Exception e)
        {
            Console.WriteLine("The file could not be read:");
            Console.WriteLine(e.Message);
        }
        for (int i = 0; i < l1.Count; i++)
        {
            int e = l1[i];
            int n = 0;

            for (int j = 0; j < l2.Count; j++)
            {
                if (l2[j] == e)
                {
                    n++;
                }
            }

            ans += n * e;
        }

        Console.WriteLine(ans);
    }
}

File: advent_of_code/2025/03/second.js
const fs = require('node:fs')

const input = fs.readFileSync('./input.txt', 'utf8')

const lines = input.split('\n')

let ans = 0

for (const line of lines) {
  let pMax = -1
  let curr = 0

  for (let n = 0; n < 12; n++) {
    curr *= 10

    let dMax = -1

    for (let i = pMax + 1; i < line.length - (11 - n); i++) {
      const d = Number(line[i])
      if (d > dMax) {
        dMax = d
        pMax = i
      }
    }

    curr += dMax
  }

  ans += curr
}


console.log(ans)